### PR TITLE
feat: custom-notes feature in webui

### PR DESCRIPTION
**Pull request by MasterK0927**
Allows users to jot down and save personal notes directly within the browser. The feature includes two main implementations: an in-toolbar button and a dedicated WebUI page.

**1. Notes Button in the Toolbar:**

- Button in the browser toolbar renders a popup when clicked.

- Popup allows users to enter and save custom notes or text.

- Notes are stored persistently across sessions using the browser's local storage.

**2. Dedicated WebUI Page:**

- Custom Notes feature has been extended to a dedicated WebUI page.

- Inkludes option for Summarising and Rephrasing the Notes. 

_API Calls are tested in two ways_ :

- Uses native JavaScript APIs to perform API calls directly. Certain checks restricting WebUI API calls are disabled to enable this.

- Backend communicates with WebUI page via Mojo IPC mechanism, delegating calls to the JavaScript frontend. API calls are routed through the browser’s C++ backend using GURL.

**Additonal Comments**

- Tested on Ubuntu 24.1
- Synced with 130

### CODE REVIEW

File: components/brave_custom_notes/resources/page/container.tsx [hunk 1]
```diff
@@ -67,38 +77,98 @@ const NoNotesMessage = styled.p`
...
+    // Fetch mojom instance
+    const customNotesAPI = getCustomNotesHandlerInstance();
+
+    // Load existing notes on component mount (from backend)
+    useEffect(() => {
+        loadNotes();
+    }, []);
+
+    const loadNotes = () => {
+        customNotesAPI.pageHandler.getAllNotes().then((response: { notes: Note[] }) => {
```
Comment (udaybansal19) [inline components/brave_custom_notes/resources/page/container.tsx]: is it possible to use async await here?

File: components/brave_custom_notes/resources/page/container.tsx [hunk 2]
```diff
@@ -67,38 +77,98 @@ const NoNotesMessage = styled.p`
...
+            setNotesList(response.notes || []);
+        });
+    };
 
     const handleAddNote = () => {
         if (note.trim()) {
-            setNotesList([...notesList, note]);
-            setNote('');
+            // Add note to the backend via mojom handler
+            customNotesAPI.pageHandler.addNote(note).then(() => {
```
Comment (udaybansal19) [inline components/brave_custom_notes/resources/page/container.tsx]: same here. using .then might introduce complications during development

**Comment by MasterK0927** [comment]
- [x]  Changing branch name to more meaningful name
- [x]  Branch is in sync with 130
- [x]  Adding more meaningful description and heading.

**Comment by MasterK0927** [comment]
Things Broken:

- [ ] IsWebUIAllowedToMakeNetworkRequests broke, need to re review it

**Comment by MasterK0927** [comment]
Migrated to #89

**Pull request closed without merging**